test/integration/scheduler/queue_test.go

```diff
@@ -303,8 +317,15 @@ func TestCustomResourceEnqueue(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Pop fake-pod out. It should be unschedulable.
-	podInfo := testCtx.Scheduler.NextPod()
+	var podInfo *framework.QueuedPodInfo
+	// NextPod() is a blocking operation. Wrap it in timeout() to avoid relying on global integration
+	// test timeout (10m) to abort.
+	if err := timeout(testCtx.Ctx, time.Second*15, func() {
```

Comment:
Maybe all these lines could be a util function too. I would call it `nextPodorFatal` and accept `t` to be able to call Fatal from it.

Reply:
SG. `nextPodOrDie` may read better.